### PR TITLE
implement to_solana for ed25519 pubkey

### PR DESCRIPTION
**Pull request by lthiery**

### CODE REVIEW

File: src/public_key.rs [hunk 1]
```diff
@@ -381,6 +381,14 @@ impl PublicKey {
             PublicKeyRepr::Secp256k1(..) => secp256k1::PUBLIC_KEY_LENGTH,
         }
     }
+
+    pub fn to_solana(&self) -> Option<solana_sdk::pubkey::Pubkey> {
```
Comment (madninja) [inline src/public_key.rs:385]: Should this not be a TryFrom on PublicKey itself so it can return errors for the other types?

**Review by madninja — COMMENTED** [review]
Could you make this optional, please? This library is used on hotspots where Solana functionality is not needed

**Comment by lthiery** [comment]
@madninja are these the changes you were looking for?

File: src/public_key.rs [hunk 2]
```diff
@@ -383,6 +383,19 @@ impl PublicKey {
...
 
+#[cfg(feature = "solana")]
+impl TryFrom<PublicKey> for solana_sdk::pubkey::Pubkey {
+    type Error = &'static str;
+
+    fn try_from(public_key: PublicKey) -> std::result::Result<Self, Self::Error> {
+        if let PublicKeyRepr::Ed25519(key) = public_key.inner {
+            Ok(solana_sdk::pubkey::Pubkey::new(key.as_ref()))
+        } else {
+            Err("Only Helium Ed25519 keys may be converted to a Solana address")
```
Comment (madninja) [inline src/public_key.rs:394]: Not quite the error I was aiming for.. Can we actually have an `DecodeError::Unsupported` enum entry instead?

Comment (lthiery) [reply src/public_key.rs]: Fixed

File: src/public_key.rs [hunk 3]
```diff
@@ -385,13 +385,13 @@ impl PublicKey {
 
 #[cfg(feature = "solana")]
 impl TryFrom<PublicKey> for solana_sdk::pubkey::Pubkey {
-    type Error = &'static str;
+    type Error = error::DecodeError;
```
Comment (madninja) [inline src/public_key.rs:388]: nope.. this should be `error::Error` or just `Error` in this case like with all the other TryFrom's there

File: src/error.rs [hunk 4]
```diff
@@ -50,6 +50,8 @@ pub enum DecodeError {
     NotCompact,
     #[error("missing type byte")]
     MissingType,
+    #[error("only Helium Ed25519 keys may be converted to a Solana key")]
+    UnsupportedSolanaConversion,
```
Comment (madninja) [inline src/error.rs:54]: Can we make this more generic? i.e. `Unsupported(String)` and pass that message in as part of the TryFrom? There'll likely be other unsupported decode error calls

Comment (lthiery) [reply src/error.rs]: is `&'static str` ok?

File: src/public_key.rs (hunk 3 above)
Comment (lthiery) [reply src/public_key.rs]: fixed. checking on CI now

**Review by madninja — APPROVED** [review]

**Pull request merged**